caom2pipe/tests/test_execute_composable.py

```diff
@@ -374,10 +219,7 @@ def test_data_execute(test_config):
         assert repo_client_mock.update.called, 'update call missed'
         assert test_observer.metrics.observe.called, 'observe not called'
     finally:
-        if os.path.exists(test_fits_fqn):
-            os.unlink(test_fits_fqn)
-        if os.path.exists(test_dir):
-            os.rmdir(test_dir)
+        _clean_up_dir(test_dir)
```

Comment:
You might want to consider `tempfile` package in order to avoid the `try/finally` and checks if directory exists etc.

Reply:
The pipelines cannot rely on `/tmp` being sized appropriately - e.g. CFHT has files that range to 80GB.